File: demof/Controllers/MyCalendarController.fs
namespace Demof

open System.Collections.Generic
open System.Threading.Tasks
open Microsoft.AspNetCore.Mvc
open System
open MediatR
open demof


[<Route("api/[controller]")>]
type MyCalendarController(mediator: IMediator) =
    inherit Controller()
    let _mediator = mediator
    [<HttpGet("{id}")>]
    member this.Get id = _mediator.Send(new GetBranchScheduleItems(id))

File: demo/Controllers/MyCalendarController.cs
using System.Collections.Generic;
using System.Threading.Tasks;
using MediatR;
using Microsoft.AspNetCore.Mvc;

namespace demo
{
    [Route("api/[controller]")]
    
    public class MyCalendarController : Controller
    {
	    private readonly IMediator _mediator;

	    public MyCalendarController(IMediator mediator)
	    {
		    _mediator = mediator;
	    }

        [HttpGet("{id}")]
        public async Task<List<AgendaScheduleItem>> Get(int id) => await _mediator.Send(new GetBranchScheduleItems { Year = id });
    }
}